docs/state-diff/README.md

```diff
@@ -0,0 +1,185 @@
+
+In short, the steps to add Foundry execution results to Kontrol proofs are the following:
+
+1. Create a fresh Foundry profile to isolate the `src` and `test` folder from the rest of the codebase.
+2. Write a contract that uses the provided Solidity files to record the desired execution and execute it with Foundry.
+3. Feed the JSON files from the last step to Kontrol to produce the necessary summary contracts for ease of proof writing.
+4. Write your symbolic property tests with the right dependencies.
+5. Execute Kontrol proofs with the right options to include the recorded computation with Foundry.
+
+## ✨ Fresh Foundry Profile ✨ 
+
+The goal of this step is to create a minimal set of dependencies to run Kontrol proofs while including all the relevant bytecode. That is, we are separating verified bytecode and the sourcecode that produced it. This keeps what is executed by Kontrol to only the essentials, allowing for faster runtimes.
+
+To achieve this separation we'll introduce a new Foundry profile, usually called `kprove`. We also need a fresh folder where the proofs will live. If, for instance, that folder is `test/kontrol/proofs`, the resulting profile would look as follows:
+```toml
+[profile.kprove]
+src = 'test/kontrol/proofs'
+out = 'kout-proofs'
+test = 'test/kontrol/proofs'
+script = 'test/kontrol/proofs'
+```
+
+### ⚠️  Special `out` directory ⚠️
+
+Note that we're also setting a different `out` directory named `kout-proofs`. Because of this, any Kontrol related command will have to be executed in the context of the `kprove` profile. This can be achieved either by `export FOUNDRY_PROFILE=kprove` or by prepending any Kontrol command with that flag (e.g., `FOUNDRY_PROFILE=kprove kontrol list`).
+
+## 🖋️ Record your execution 🖋️
+
+To record your execution and save its output to a JSON file, simply use the modifier `recordStateDiff` in the [`test/kontrol/state-diff/record-state-diff/RecordStateDiff.sol`](./test/kontrol/state-diff/record-state-diff/RecordStateDiff.sol) file. That is, the initial set up of your proofs has to be run in a function with the `recordStateDiff` modifier. An example of this can be found in [`test/kontrol/state-diff/proof-initialization.sol`](test/kontrol/state-diff/proof-initialization.sol).
+
+On top of that, the `save_address` function allows to save the name of the deployed contract and their addresses into a separate JSON file. This will be crucial to easily write the symbolic property tests later on.
+
+### 🏷️ Name the files 🏷️
+
+There are two different JSON files that can be created, one containing the recorded state updates and another one with the saved names of important addresses. On top of that, we have to define in which directory do these files live. For simplicity, we've made it so that these parameters are set with environment variables:
```

Comment:
Maybe I'd clarify what the "important addresses" are — the ones that are meant to be used in the tests?

Reply:
Good catch. Gotta explain what these VIP addresses are!